### PR TITLE
Fix parsing of largest negative bigint value

### PR DESCRIPTION
**Pull request by martint**
The largest negative bigint value (i.e., Long.MIN_VALUE) was previously being parsed as
-(9223372036854775808), which is a number too big to fit in a java long (the range is
[-9223372036854775808, 9223372036854775807]).  This change makes the sign part of the
number, so it is now parsed and stored properly in the syntax tree.

We also update parsing of decimal and double numbers to avoid tokens such as -123E45
or -123.45 from being interpreted as an integer (-123) with another token trailing
behind it.

Before:
```
presto> SELECT -9223372036854775808;
Query 20190522_012223_00016_zcjux failed: line 1:1: Invalid numeric literal: 9223372036854775808
```
After:

```
presto> SELECT -9223372036854775808;
        _col0
----------------------
 -9223372036854775808
(1 row)
```

### CODE REVIEW

**Comment by martint** [comment]
@electrum, I had to take a different approach. Making the `-` sign part of the tokens introduces some ambiguities with subtractions (e.g., `1-2` tokenizes as `<int> <int>`). Please review again.

**Comment by martint** [comment]
BTW, another option would've been to allow LongLiteral to hold a value of any size (we might still want to do this in the future, but as a way to represent integer literals in general, before their actual type has been established), but it would still cause trouble during type analysis. The expression would be UnaryArithmetic('-', LongLiteral('9223372036854775808')), but since `9223372036854775808` is not a valid bigint, its type would automatically be determined to be `decimal(10)`, which is not what we want.

File: presto-main/src/test/java/io/prestosql/sql/TestExpressionInterpreter.java [hunk 1]
```diff
@@ -1507,8 +1507,8 @@ private static Object evaluate(String expression)
     private static void assertRoundTrip(String expression)
     {
         ParsingOptions parsingOptions = createParsingOptions(TEST_SESSION);
-        assertEquals(SQL_PARSER.createExpression(expression, parsingOptions),
-                SQL_PARSER.createExpression(formatExpression(SQL_PARSER.createExpression(expression, parsingOptions), Optional.empty()), parsingOptions));
+        Expression parsed = SQL_PARSER.createExpression(expression, parsingOptions);
```
Comment (electrum) [inline presto-main/src/test/java/io/prestosql/sql/TestExpressionInterpreter.java]: Maybe introduce another variable so the `createExpression` calls look the same:
```java
Expression parsed = SQL_PARSER.createExpression(expression, parsingOptions);
String formatted = formatExpression(parsed, Optional.empty());
assertEquals(parsed, SQL_PARSER.createExpression(formatted, parsingOptions));
```
Also move this to another commit since it's just test cleanup

Comment (martint) [reply presto-main/src/test/java/io/prestosql/sql/TestExpressionInterpreter.java]: Hah! I had that exact code in one of my work-in-progress patches. I’ll update it.

Btw, it is already a separate commit
